### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by craig-riecke**
This adds all the unit tests I've written so far, with the exception of OpenFlow0x04 unit tests, which must be cherry picked from the openflow13 branch.

### CODE REVIEW

**Comment by craig-riecke** [comment]
Also added a bunch of documentation to mli files, removed functions that were no longer used (and unlikely to be used in the future).

File: lib_test/Test_Frenetic_NetKAT.ml [hunk 1]
```diff
@@ -1,575 +1,5 @@
...
-  let prop_compile_ok (p, q, r, pkt) =
-    not (compare_compiler_output (Union(Union(p, Seq (q, r)), q)) q pkt)
-    ||  (compare_compiler_output (Union(Seq(p, Star r), q)) q pkt) in
-  check gen_pol_3 prop_compile_ok
-
-let%test "quickcheck ka-lfp-r eval" =
-  let prop_compile_ok (p, q, r, pkt) =
-    not (compare_eval_output (Union(Union(p, Seq (q, r)), q)) q pkt)
-    ||  (compare_eval_output (Union(Seq(p, Star r), q)) q pkt) in
-  check gen_pol_3 prop_compile_ok
```
Comment (smolkaj) [inline lib_test/Test_Frenetic_NetKAT.ml]: Did these get moved somewhere else?

**Comment by smolkaj** [comment]
@craig-riecke: I allowed myself to add a commit to this pull-request (it seemed to fit in well).

File: lib_test/Test_Frenetic_NetKAT.ml (hunk 1 above)
Comment (craig-riecke) [inline lib_test/Test_Frenetic_NetKAT.ml]: My mistake.  They seemed to be more appropriate in Test_Frenetic_Compiler, so I was moving them from Test_Frenetic_NetKat, but dropped them on the way over.  Will fix...

**Comment by craig-riecke** [comment]
@smolkaj Nice ... that change makes the tests less brittle. The Exn.does_raise is a good touch as well - I'll use that in further unit tests.

**Pull request merged**